### PR TITLE
init kuyen-icon-theme, and antu-icon-theme

### PR DESCRIPTION
**Pull request by iamanaws**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/data/icons/antu-icon-theme/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenvNoCC, fetchFromGitLab, gtk3 }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "antu-icon-theme";
+  version = "ff16444b";
```
Comment (Janik-Haag) [inline pkgs/data/icons/antu-icon-theme/default.nix:5]: the version should be unstable-$date_of_last_upstream_commit if you use the rev

File: pkgs/data/icons/kuyen-icon-theme/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  dontDropIconThemeCache = true;
+
+  installPhase = ''
+    mkdir -p $out/share/icons/kuyen-icons
+    cp -r * $out/share/icons/kuyen-icons
+  '';
+
+  meta = with lib; {
+    description = "A colourful flat theme designed for Plasma desktop";
```
Comment (Janik-Haag) [inline pkgs/data/icons/kuyen-icon-theme/default.nix:24]:
```suggestion
    description = "A colourful flat theme designed for Plasma Desktop";
```
would be nice to have a consistent capitalization

File: pkgs/data/icons/antu-icon-theme/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitLab {
+    owner = "froodo_alexis";
+    repo = "antu-icons";
+    rev = version;
+    sha256 = "sha256-CLcr+X/b0moVEBV0O/dzCDq4w5G2+KRLUBdqKm0eAKA=";
```
Comment (Janik-Haag) [inline pkgs/data/icons/antu-icon-theme/default.nix:11]:
```suggestion
    hash = "sha256-CLcr+X/b0moVEBV0O/dzCDq4w5G2+KRLUBdqKm0eAKA=";
```
hash is the more modern thing to use.

File: pkgs/data/icons/kuyen-icon-theme/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+stdenvNoCC.mkDerivation rec {
+  pname = "kuyen-icon-theme";
+  version = "747e0e01";
+
+  src = fetchFromGitLab {
+    owner = "froodo_alexis";
+    repo = "kuyen-icons";
+    rev = version;
+    sha256 = "sha256-6BK6+6CZhcPgDNuBIF9cYXGwS414NlL+KvRdsDzziaQ=";
```
Comment (Janik-Haag) [inline pkgs/data/icons/kuyen-icon-theme/default.nix:11]: same here with version and hash

File: pkgs/data/icons/antu-icon-theme/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  nativeBuildInputs = [ gtk3 ];
```
Comment (Janik-Haag) [inline pkgs/data/icons/antu-icon-theme/default.nix:14]: why are you including gtk3?

File: pkgs/data/icons/antu-icon-theme/default.nix (hunk 1 above)
Comment (h7x4) [reply pkgs/data/icons/antu-icon-theme/default.nix:5]: I believe guidelines were updated recently, and unstable without previous version tags should be prefixed with `0-` (i.e. `0-unstable-YYYY-MM-DD`): https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

Comment (Janik-Haag) [reply pkgs/data/icons/antu-icon-theme/default.nix:5]: uh til, thanks ^^

Comment (Iamanaws) [reply pkgs/data/icons/antu-icon-theme/default.nix:5]: Like this?
`version = "0-unstable-2024-01-23";
`

`rev = "279be277";`

File: pkgs/data/icons/antu-icon-theme/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    rev = "279be277";
+    hash = "sha256-CLcr+X/b0moVEBV0O/dzCDq4w5G2+KRLUBdqKm0eAKA=";
+  };
+
+  dontDropIconThemeCache = true;
+
+  installPhase = ''
+    mkdir -p $out/share/icons/antu-icons
+    cp -r * $out/share/icons/antu-icons
+  '';
```
Comment (h7x4) [inline pkgs/data/icons/antu-icon-theme/default.nix:19]: [Don't forget the to run the pre/post hooks :)](https://nixos.org/manual/nixpkgs/unstable/#sec-stdenv-phases)

```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -p $out/share/icons/antu-icons
    cp -r * $out/share/icons/antu-icons
    
    runHook postInstall
  '';
```

File: pkgs/data/icons/antu-icon-theme/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+  meta = with lib; {
+    description = "A smooth icon theme designed for Plasma Desktop";
+    homepage = "https://gitlab.com/froodo_alexis/Antu-icons";
```
Comment (h7x4) [inline pkgs/data/icons/antu-icon-theme/default.nix:23]:
```suggestion
    homepage = "https://gitlab.com/froodo_alexis/Antu-icons";
    changelog = "https://gitlab.com/froodo_alexis/Antu-icons/-/releases";
```

File: pkgs/data/icons/kuyen-icon-theme/default.nix [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  dontDropIconThemeCache = true;
+
+  installPhase = ''
+    mkdir -p $out/share/icons/kuyen-icons
+    cp -r * $out/share/icons/kuyen-icons
+  '';
+
+  meta = with lib; {
+    description = "A colourful flat theme designed for Plasma Desktop";
+    homepage = "https://gitlab.com/froodo_alexis/kuyen-icons";
```
Comment (h7x4) [inline pkgs/data/icons/kuyen-icon-theme/default.nix:23]:
```suggestion
    homepage = "https://gitlab.com/froodo_alexis/kuyen-icons";
    changelog = "https://gitlab.com/froodo_alexis/kuyen-icons/-/releases";
```

File: pkgs/top-level/all-packages.nix [hunk 9]
```diff
@@ -29028,6 +29028,8 @@ with pkgs;
 
   ant-nebula-theme = callPackage ../data/themes/ant-theme/ant-nebula.nix { };
 
+  antu-icon-theme = callPackage ../data/icons/antu-icon-theme { };
```
Comment (h7x4) [inline pkgs/top-level/all-packages.nix:29031]: I believe these would have to go to [`pkgs/by-name`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md) now.

File: pkgs/data/icons/antu-icon-theme/default.nix [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    license = licenses.cc-by-nc-sa-30;
+    platforms = platforms.linux;
```
Comment (h7x4) [inline pkgs/data/icons/antu-icon-theme/default.nix:25]: I think icons can be useful anywhere, even though things like `index.theme` might target linux-based graphical systems.

```suggestion
    platforms = platforms.all;
```

File: pkgs/data/icons/kuyen-icon-theme/default.nix [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    license = licenses.cc-by-nc-sa-30;
+    platforms = platforms.linux;
```
Comment (h7x4) [inline pkgs/data/icons/kuyen-icon-theme/default.nix:25]:
```suggestion
    platforms = platforms.all;
```

File: pkgs/data/icons/antu-icon-theme/default.nix [hunk 12]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenvNoCC, fetchFromGitLab }:
+
+stdenvNoCC.mkDerivation rec {
```
Comment (h7x4) [inline pkgs/data/icons/antu-icon-theme/default.nix:3]:
```suggestion
stdenvNoCC.mkDerivation {
```

File: pkgs/data/icons/kuyen-icon-theme/default.nix [hunk 13]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenvNoCC, fetchFromGitLab }:
+
+stdenvNoCC.mkDerivation rec {
```
Comment (h7x4) [inline pkgs/data/icons/kuyen-icon-theme/default.nix:3]:
```suggestion
stdenvNoCC.mkDerivation {
```

**Comment by h7x4** [comment]
Could you squash the changes into 3 commits like this?

```
maintainers: add iamanaws
kuyen-icon-theme: init at 0-unstable-2024-01-27
antu-icon-theme: init at 0-unstable-2024-01-23
```

If you're unfamiliar with interactive rebasing, feel free to ask. It might be a good idea to mark the PR as draft before force pushing. If you accidentally include a bunch of other commits in the rebase, it will ping a ton of people unless it's marked as draft.

**Comment by Janik-Haag** [comment]
@Iamanaws squashing the package related commits is still a todo and the pipeline is also still failing because of formatting. It seems like you are only using the github webinterface, I can highly recommend you to use some texteditor like vscode or neovim locally. Feel free to ask for help if you need some pointers.

**Pull request closed without merging**